pkg/ddevapp/dotddev_assets/providers/pantheon.yaml

```diff
@@ -67,13 +63,16 @@ auth_command:
     terminus auth:login --machine-token="${TERMINUS_MACHINE_TOKEN}" || ( echo "terminus auth login failed, check your TERMINUS_MACHINE_TOKEN" && exit 1 )
     terminus aliases 2>/dev/null
 
+#make sure env is awake, and grab a fresh DB export
 db_pull_command:
   command: |
     # set -x   # You can enable bash debugging output by uncommenting
     set -eu -o pipefail
     echo "Using PANTHEON_SITE=${PANTHEON_SITE} PANTHEON_ENVIRONMENT=${PANTHEON_ENVIRONMENT}"
     pushd /var/www/html/.ddev/.downloads >/dev/null
-    terminus backup:get ${PANTHEON_SITE}.${PANTHEON_ENVIRONMENT} --element=db --to=db.sql.gz
+    terminus env:wake -- ${PANTHEON_SITE}.${PANTHEON_ENVIRONMENT}
+    FRESH_DB_DUMP_STRING=$(terminus connection:info "${PANTHEON_SITE}.${PANTHEON_ENVIRONMENT}" --field=mysql_command | sed 's,^mysql,mysqldump --no-autocommit --single-transaction --opt -Q,')
+    eval "$FRESH_DB_DUMP_STRING | gzip > db.sql.gz" > /dev/null 2>&1
```

Comment:
I'm not sure about this change, it's very slow even for a small database and you don't see any useful progress.

I would prefer:

```
echo "Waking up the project environment..."
terminus env:wake -- ${PANTHEON_SITE}.${PANTHEON_ENVIRONMENT}
...
echo "Running mysqldump on the live database..."
...
```

And maybe it should be optional by default. Some new `env` can be introduced:

```
ddev pull pantheon --environment=PANTHEON_PULL_NO_BACKUP=true
```

Reply:
I see.
Then at least an `echo` can help with understanding what commands are executed while waiting.